wvd-templates/DSC/Functions.ps1

```diff
@@ -327,4 +330,163 @@ function IsRDAgentRegistryValidForRegistration {
     return @{
         result = $true
     }
+}
+
+function RunMsiWithRetry {
+    param(
+        [Parameter(mandatory = $true)]
+        [string]$programDisplayName,
+
+        [Parameter(mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [String[]]$argumentList, #Must have at least 1 value
+
+        [Parameter(mandatory = $true)]
+        [string]$msiOutputLogPath,
+
+        [Parameter(mandatory = $false)]
+        [switch]$isUninstall,
+
+        [Parameter(mandatory = $false)]
+        [switch]$msiLogVerboseOutput
+    )
+    Set-StrictMode -Version Latest
+
+    if ($msiLogVerboseOutput) {
+        $argumentList += "/l*vx $msiOutputLogPath" 
+    }
+    else {
+        $argumentList += "/l* $msiOutputLogPath"
+    }
+
+    $retryTimeToSleepInSec = 30
+    $retryCount = 0
+    do {
+        $modeAndDisplayName = ($(if ($isUninstall) { "Uninstalling" } else { "Installing" }) + " $programDisplayName")
+
+        if ($retryCount -gt 0) {
+            Write-Log -Message "Retrying $modeAndDisplayName in $retryTimeToSleepInSec seconds because it failed with Exit code=$sts This will be retry number $retryCount"
+            Start-Sleep -Seconds $retryTimeToSleepInSec
+        }
+
+        Write-Log -Message ( "$modeAndDisplayName" + $(if ($msiLogVerboseOutput) { " with verbose msi logging" } else { "" }))
+
+
+        $processResult = Start-Process -FilePath "msiexec.exe" -ArgumentList $argumentList -Wait -Passthru
+        $sts = $processResult.ExitCode
+
+        $retryCount++
+    } 
+    while ($sts -eq 1618 -and $retryCount -lt 20) # Error code 1618 is ERROR_INSTALL_ALREADY_RUNNING see https://docs.microsoft.com/en-us/windows/win32/msi/-msiexecute-mutex .
+
+    if ($sts -eq 1618) {
+        Write-Log -Message "Stopping retries for $modeAndDisplayName. The last attempt failed with Exit code=$sts"

```

Comment:
Might be helpful to log a description of this error code, like `ERROR_INSTALL_ALREADY_RUNNING`